### PR TITLE
[Improve][Zeta] Statistics server job and system resource usage #3980 

### LINKED ISSUE #3980: [Improve][Zeta] Statistics server job and system resource usage
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Since Zeta currently lacks access to system and task execution statistics, this is not user-friendly. Jobs are completely black boxes to users at runtime.Therefore, we need to count job execution and system resource usage.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liugddx**
close #3980

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by 2000liux** [comment]
nice

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java [hunk 1]
```diff
@@ -519,24 +524,129 @@ public void printExecutionInfo() {
         int poolSize = threadPoolExecutor.getPoolSize();
         long completedTaskCount = threadPoolExecutor.getCompletedTaskCount();
         long taskCount = threadPoolExecutor.getTaskCount();
+
+        AtomicLong createdJobCount = new AtomicLong();
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java:545]: I think maybe we should change the time to print the info. If only print when job done, streaming job will never print this info.

Comment (liugddx) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java:528]: I understand that this is a server-side tasks regularly, have nothing to do with specific jobs.

**Comment by liugddx** [comment]
image

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java (hunk 1 above)
Comment (liugddx) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java:545]: > I think maybe we should change the time to print the info. If only print when job done, streaming job will never print this info.

Done.

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java [hunk 2]
```diff
@@ -141,6 +148,8 @@ public CoordinatorService(@NonNull NodeEngineImpl nodeEngine, @NonNull SeaTunnel
         this.engineConfig = engineConfig;
         masterActiveListener = Executors.newSingleThreadScheduledExecutor();
         masterActiveListener.scheduleAtFixedRate(this::checkNewActiveMaster, 0, 100, TimeUnit.MILLISECONDS);
+        monitorJob = Executors.newSingleThreadScheduledExecutor();
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java]: Hi, we already have `monitorService` in SeaTunnelServer.java, please use it to printJobDetailInfo. Don't create new ScheduledExecutor. Judging that if it is the master node, just print the corresponding information in `printExecutionInfo`.

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java [hunk 3]
```diff
@@ -120,13 +126,14 @@ public class CoordinatorService {
     /**
      * If this node is a master node
      */
+    @Getter
```
Comment (Hisoka-X) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/CoordinatorService.java]: Just use `CoordinatorService::isCoordinatorActive`.

**Review by EricJoy2048 — COMMENTED** [review]
Please add release note to `/release-note.md`

**Comment by EricJoy2048** [comment]
> image

I'm worried that if there are many slots in a single node, will the log output here take up too much space? Will this affect the problem we define through the log?

**Comment by liugddx** [comment]
> slots
I agree. I will provide other ways to get system resources.

**Comment by liugddx** [comment]
image

**Review by Hisoka-X — DISMISSED** [review]

**Comment by liugddx** [comment]
please wait #4028

**Review by EricJoy2048 — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**